react/week5/githubusersprofile/src/Components/Profile.js

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import "./Profile.css";
+
+export function Profile({ user }) {
+  const [loadedUser, setLoadedUser] = useState(user);
+  const { userName } = useParams();
+
+  //useState is for tracking user's properties and their changes.
+  //useParams returns an object of key/value pairs of URL parameters, in ourcase is "userName"
+  //param is loaded inside profile component, hence its imported in this component.
+  //useEffect is for fetching.
+
+  useEffect(() => {
+    if (user && user.id) return; //if there is user and has user.id, then fetch his/her details from github API.
+
```

Comment:
```suggestion
if (userName) {
```
fetch data if you have userName